src/commonmark/en/content/developer/web-api.md

```diff
@@ -16464,6 +16476,236 @@ based on data element
+    "assignedUsers" : ["a3kGcGDCuk7", "a3kGcGDCuk8"],
+    "followUp": false,
+    "trackedEntityInstance": "a3kGcGDCuk6",
+    "events": ["a3kGcGDCuk7", "a3kGcGDCuk8"],
+    "fields": "eventDate,dueDate",
+    "order": "dueDate:asc,createdDate:desc"
+  }
+</td>
+</tr>
+</tbody>
+</table>
+
+<table>
+<caption>Event Query Criteria definition</caption>
+<colgroup>
+<col style="width: 33%" />
+<col style="width: 33%" />
+<col style="width: 33%" />
+</colgroup>
+<tbody>
+<tr class="odd">
+<td>followUp</td>
+<td>Used to filter events based on enrollment followUp flag. Possible values are true|false.</td>
+<td>"followUp": true</td>
+</tr>
+<tr class="even">
+<td>ouMode</td>
+<td>To specify the OU selection mode. Possible values are SELECTED| CHILDREN|DESCENDANTS|ACCESSIBLE|CAPTURE|ALL</td>
+<td>"ouMode": "SELECTED"</td>
+</tr>
+<tr class="odd">
+<td>assignedUserMode</td>
+<td>To specify the assigned user selection mode. Possible values are CURRENT| PROVIDED | NONE | ANY </td>
+<td>"assignedUserMode": "NONE"</td>
+</tr>
+<tr class="even">
+<td>assignedUsers</td>
+<td>To specify a list of assigned users for events.</td>
+<td>"assignedUsers": ["a3kGcGDCuk7", "a3kGcGDCuk8"]</td>
+</tr>
+<tr class="odd">
+<td>trackedEntityInstance</td>
+<td>To specify a tracked entity instance that has to be used when filtering events</td>
+<td>"trackedEntityInstance": "A03MvHHogjR"</td>
+</tr>
+<tr class="even">
+<td>order</td>
+<td>To specify ordering/sorting of fields and its directions</td>
+<td>"order"="a3kGcGDCuk6:desc,eventDate:asc"</td>
+</tr>
+<tr class="odd">
+<td>filters</td>
+<td>To specify filters to be applied when listing events</td>
+<td>"filters"=["a3kGcGDCuk6:LT:4"]</td>
+</tr>
+<tr class="even">
+<td>fields</td>
+<td>To specify fields that has to be part of the event listing response.</td>
+<td>"fields"="a3kGcGDCuk6,eventDate,dueDate"</td>
+</tr>
+<tr class="odd">
+<td>dataElements</td>
+<td>To specify dataElements that has to be part of the event grid query response.</td>
+<td>"dataElements"=["a3kGcGDCuk6"]</td>
+</tr>
+<tr class="even">
+<td>status</td>
+<td> Any valid EventStatus</td>
+<td>  "eventStatus": "COMPLETED"</td>
+</tr>
+<tr class="odd">
+<td>createdDate</td>
+<td>DatePeriod object containing a from and to date for filtering based on created date.</td>
+<td>{ &quot;from&quot;:&quot;2018-01-01&quot; , &quot;to&quot;: &quot;2019-01-01&quot;}</td>
+</tr>
+<tr class="even">
+<td>dueDate</td>
+<td>DatePeriod object containing a from and to date for filtering based on due date.</td>
+<td>{ &quot;from&quot;:&quot;2018-01-01&quot; , &quot;to&quot;: &quot;2019-01-01&quot;}</td>
+</tr>
+<tr class="odd">
+<td>lastUpdatedDate</td>
+<td>DatePeriod object containing a from and to date for filtering based on last updated date.</td>
+<td>{ &quot;from&quot;:&quot;2018-01-01&quot; , &quot;to&quot;: &quot;2019-01-01&quot;}</td>
+</tr>
+<tr class="even">
+<td>events</td>
+<td>To specify list of events</td>
+<td>"events"=["a3kGcGDCuk6"]</td>
+</tr>
+</tbody>
```

Comment:
We need eventDate and completedDate filters also

Reply:
And as commented earlier, we need relative periods here. The trackedEntityInstance filters has a syntax that can be used as inspiration.

Reply:
1. isnt eventDate same as createdDate?
2. I can add completedDate in the filter definition. But the event listing APIs currently does not support filtering based on completedDate.

Reply:
1. Both a created date and event date exists for an event - The event date is the date we usually reference and use for filtering.
2. What dates are supported for this filtering at the moment?

Reply:
In the api, we have 3 set of dates. 1) startDate/endDate

Reply:
ok great - lets document how to use these for now.